File: Exercises/answer_keys/python/building_ladder.py
# This program has the goal to print a "Ladder" of chars that 
# is chosen by the user. The height of the "Ladder" also
# is chosen by the user.

def ladder(h, d):

    h = int(h)
    tmp = 1

    for i in range(h):
        for j in range(tmp):
            print (d, end= ' ')
        print("")
        tmp += 1

def main():

    height, digit = input("Input a height value and a type of digit: ").split()

    ladder(height, digit)

if __name__ == "__main__":
    main()

File: Exercises/answer_keys/cpp/building_ladder.cpp
// This program has the goal to print a "Ladder" of chars that 
// is chosen by the user. The height of the "Ladder" also
// is chosen by the user.

#include <iostream>

using namespace std;

void ladder(int h, char d){
    for(int i = 1; i <= h; i++){
        for(int k = 0; k < i; k++){
            cout << d << " ";
        }
        cout << endl;
    }
}

int main(){
    int height;
    char digit;

    cout << "Input a height value and a type of digit: ";
    cin >> height;
    cin >> digit;

    cout << endl; // just skips a line for better visualization

    ladder(height, digit);
}